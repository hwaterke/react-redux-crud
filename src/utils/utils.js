// @flow

export function invariant(condition: any, message: string) {
  if (!condition) {
    const error = new Error(message)
    error.name = 'Invariant Violation'
    throw error
  }
}
