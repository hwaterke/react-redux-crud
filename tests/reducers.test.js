// @flow
import reduxCrud from 'redux-crud/dist/index'
import {Book, initialiseStore} from './utils'

describe('Reducers', () => {
  let store

  beforeEach(() => {
    store = initialiseStore()
  })

  test('should have a correct initial state', function() {
    const state = store.getState()
    expect(state).toMatchSnapshot()
  })

  test('should have a correct state after FETCH_START', function() {
    const actionTypes = reduxCrud.actionTypesFor(Book.name)
    store.dispatch({type: actionTypes.fetchStart})

    expect(store.getState()).toMatchSnapshot()
  })

  test('should have a correct state after FETCH_SUCCESS', function() {
    const actionTypes = reduxCrud.actionTypesFor(Book.name)
    store.dispatch({type: actionTypes.fetchStart})
    store.dispatch({
      type: actionTypes.fetchSuccess,
      records: [{uuid: 1, name: 'Name'}],
    })

    expect(store.getState()).toMatchSnapshot()
  })
})
