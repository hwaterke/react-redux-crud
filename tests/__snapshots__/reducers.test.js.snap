// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers should have a correct initial state 1`] = `
Object {
  "lastAction": Object {
    "type": "@@redux/INIT",
  },
  "resources": Object {
    "books": Object {},
  },
  "resourcesActivity": Object {
    "books": Object {
      "creating": Object {},
      "fetching": Object {},
      "fetchingAll": 0,
      "removing": Object {},
      "updating": Object {},
    },
  },
}
`;

exports[`Reducers should have a correct state after FETCH_START 1`] = `
Object {
  "lastAction": Object {
    "type": "BOOKS_FETCH_START",
  },
  "resources": Object {
    "books": Object {},
  },
  "resourcesActivity": Object {
    "books": Object {
      "creating": Object {},
      "fetching": Object {},
      "fetchingAll": 1,
      "removing": Object {},
      "updating": Object {},
    },
  },
}
`;

exports[`Reducers should have a correct state after FETCH_SUCCESS 1`] = `
Object {
  "lastAction": Object {
    "records": Array [
      Object {
        "name": "Name",
        "uuid": 1,
      },
    ],
    "type": "BOOKS_FETCH_SUCCESS",
  },
  "resources": Object {
    "books": Object {
      "1": Object {
        "name": "Name",
        "uuid": 1,
      },
    },
  },
  "resourcesActivity": Object {
    "books": Object {
      "creating": Object {},
      "fetching": Object {},
      "fetchingAll": 0,
      "removing": Object {},
      "updating": Object {},
    },
  },
}
`;
